### PR TITLE
http_caldav_sched.c: use actual calendar owner id in IMIP notifications

### PR DESCRIPTION
**Pull request by ksmurchison**

### CODE REVIEW

File: imap/jmap_calendar.c [hunk 1]
```diff
@@ -7979,7 +7979,7 @@ static int jmap_calendarevent_participantreply(struct jmap_req *req)
         break;
     }
 
-    sched_request(req->accountid, NULL, organizer,
+    sched_request(req->accountid, req->accountid, NULL, organizer,
```
Comment (rsto) [inline imap/jmap_calendar.c:7982]: Same, the schedule_id being req->accountid looks suspicious to me.

File: imap/http_caldav.c [hunk 2]
```diff
@@ -1528,22 +1528,25 @@ static int caldav_delete_cal(struct transaction_t *txn,
         /* XXX - after legacy records are gone, we can strip this and just not send a
          * cancellation if deleting a record which was never replied to... */
 
-        char *userid = mboxname_to_userid(txn->req_tgt.mbentry->name);
+        char *cal_ownerid = mboxname_to_userid(txn->req_tgt.mbentry->name);
+        char *sched_userid = (txn->req_tgt.flags == TGT_DAV_SHARED) ?
```
Comment (rsto) [inline imap/http_caldav.c:1532]: I suggest to not allow sched_userid being NULL, even if that makes handling the allocated strings here a bit more complicated.

First, it was not clear to me when I saw this PR if sched_userid is nullable and most of the changes in this PR do not pass NULL, even if both cal_ownerid and sched_userid point to the same memory.

Second, I prefer we force us to always think about the schedule id when we call the scheduling layer. JMAP Calendars team mode calendars will require us to do anyway, so it's better to be explicit about it at all call sites.

**Review by rsto — CHANGES_REQUESTED** [review]
Looks mostly good to me, but I'd prefer sched_userid to be mandatory and some of the JMAP calls might not work for the shared calendars.

File: imap/jmap_calendar.c [hunk 3]
```diff
@@ -4111,7 +4111,7 @@ static int setcalendarevents_schedule(const char *sched_userid,
                 }
             }
             if (!omit_reply && strarray_size(schedule_addresses))
-                sched_reply(sched_userid, schedule_addresses,
+                sched_reply(sched_userid, sched_userid, schedule_addresses,
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/jmap_calendar.c:4114]: @rsto sched_userid appears to be is set to req->accountid everywhere in jmap_calendar.c.  Is this correct?  Do we want to separate the two.

File: imap/jmap_calendar.c (hunk 1 above)
Comment (ksmurchison) [reply imap/jmap_calendar.c:7982]: Fixed

File: imap/jmap_backup.c [hunk 4]
```diff
@@ -926,7 +926,7 @@ static int do_scheduling(jmap_req_t *req,
 
     if (strarray_find_case(schedule_addresses, organizer, 0) >= 0) {
         /* Organizer scheduling object resource */
-        sched_request(req->userid, schedule_addresses, organizer,
+        sched_request(req->userid, req->userid, schedule_addresses, organizer,
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/jmap_backup.c:929]: Fixed

File: imap/jmap_backup.c [hunk 5]
```diff
@@ -957,7 +957,7 @@ static int do_scheduling(jmap_req_t *req,
         }
 
         if (!omit_reply && strarray_size(schedule_addresses))
-            sched_reply(req->userid, schedule_addresses,
+            sched_reply(req->userid, req->userid, schedule_addresses,
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/jmap_backup.c:960]: Fixed

File: imap/jmap_calendar.c [hunk 6]
```diff
@@ -7941,7 +7941,7 @@ static int jmap_calendarevent_participantreply(struct jmap_req *req)
     }
 
     /* Create and send the reply */
-    sched_reply(req->accountid, &schedule_addr,
+    sched_reply(req->accountid, req->accountid, &schedule_addr,
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/jmap_calendar.c:7944]: Fixed

File: imap/http_caldav.c (hunk 2 above)
Comment (ksmurchison) [reply imap/http_caldav.c:1532]: Should be fixed

File: imap/jmap_calendar.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (rsto) [reply imap/jmap_calendar.c:4114]: right now we want the scheduling id always be the account id (=secretary mode). but we might want to change that later for team mode calendars.

**Review by rsto — APPROVED** [review]

**Pull request merged**